### PR TITLE
[DFC API] Add custom OFN product id to DFC SuppliedProduct

### LINKED ISSUE #11229: Represent OFN Product on DFC API (not just variants)
## Description

Our integration with the Data Food Consortium (DFC) standard maps OFN Variants to DFC SuppliedProducts. The DFC doesn't have an obvious model for OFN Products though. Currently, we see only variants without their product relationship on the DFC API and we can only create one variant per product.

One modelling idea was to use the DFC FunctionalProduct -> TechnicalProduct >SuppliedProduct hierarchy but the DFC is planning another implementation. From @RaggedStaff:

> An additional (self-referential) relationship will be added to Supplied Product to support Variant mappings.

I don't quite know yet what that means but it looks like we are blocked until this happens unless we want to implement an intermediate solution first.

## Acceptance Criteria & Tests

1. The SuppliedProduct endpoint allows to identify SuppliedProducts belonging together like OFN variants belong to an OFN product.
2. The SuppliedProduct endpoint allows to read and write the name and description of an OFN product.

### PR DESCRIPTION
**Pull request by mkllnk**
ℹ️ This is a funded feature. Please track your time in `#9170 OFN DFC Products`.

#### What? Why?

- Closes #11229  

The DFC doesn't model grouping of several products (variants). So we add an OFN custom property for our own integrations. Other software should ignore it. For other semantic web tools the `ofn:` URI is not defined and can't be resolved. We haven't defined our own ontology. But I don't see a need for that at the moment.

#### What should we test?

- Specs only

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: lib/data_food_consortium/connector/importer.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
+require "data_food_consortium/connector/connector"
```
Comment (pedrocarmona) [inline lib/data_food_consortium/connector/importer.rb:3]:
```suggestion
require "datafoodconsortium/connector"
```

Comment (pedrocarmona) [reply lib/data_food_consortium/connector/importer.rb:3]: well, maybe not the best change :P will read better

**Comment by rioug** [comment]
@mkllnk  I am assuming this need to be tracked agains : `#9170 OFN DFC Products` ?

**Comment by mkllnk** [comment]
@rioug 

>  I am assuming this need to be tracked agains : #9170 OFN DFC Products ?

Yes, that's right. I forgot to mention that.

File: engines/dfc_provider/lib/data_food_consortium/connector/importer.rb [hunk 2]
```diff
@@ -127,11 +127,14 @@ def skos_concept(object)
       end
 
       def guess_setter_name(predicate)
-        fragment = predicate.fragment
+        name =
```
Comment (dacook) [inline engines/dfc_provider/lib/data_food_consortium/connector/importer.rb:131]: Breaking a command over empty lines feels wrong to me, so what you think about removing the empty lines?

```suggestion
```

**Review by dacook — APPROVED** [review]
Great job! I have a small comment about formatting, but that is all.

File: engines/dfc_provider/lib/data_food_consortium/connector/importer.rb (hunk 2 above)
Comment (mkllnk) [reply engines/dfc_provider/lib/data_food_consortium/connector/importer.rb]: Okay, done. I was wondering that, too. There's another style rule to have blank lines above comments but rubocop doesn't complain here. So all good.

**Review by dacook — APPROVED** [review]

**Review by rioug — APPROVED** [review]
Looks good :+1:

**Comment by rioug** [comment]
Merging as no manual test needed.

**Pull request merged**